### ISSUE TITLE
[Bug]: `MyDataObject::getByField($value, limit: 1)` doesn't return a single object (when named parameters are used)

### ISSUE THREAD

**Issue by youwe-petervanderwal** [pimcore/pimcore#17108]
### Pimcore version

11.2.3

### Steps to reproduce

For example on the demo environment
```
// Returns a Manufacturer
$manufacturer = Manufacturer::getByName('Ford', 1);

// Returns a Manufacturer\Listing
$manufacturer = Manufacturer::getByName('Ford', limit: 1);
```

### Actual Behavior

```
// Returns a Manufacturer\Listing
$manufacturer = Manufacturer::getByName('Ford', limit: 1);
```

### Expected Behavior

```
// Returns a ?Manufacturer
$manufacturer = Manufacturer::getByName('Ford', limit: 1);

// Similar as does
$manufacturer = Manufacturer::getByName('Ford', 1);
```

**Comment by rahul-pim-dev** [comment]
Hi @youwe-petervanderwal 

In this example, you have utilized the named parameters feature introduced in PHP version 8. 
Within Pimcore, this process involves an internal call to the __callStatic method, as seen in the following link: [Pimcore Github](https://github.com/pimcore/pimcore/blob/cc1a1fff78686423f616c7fe7c5d634a2ac71600/models/DataObject/Concrete.php#L537).

This method is designed to return a single record. However, due to a limitation in PHP, [Magic methods does not support named parameters](https://github.com/php/php-src/issues/11348).

**Comment by jdreesen** [comment]
Your link just shows a bug in PHP that occurs when you create a closure for a magic static method and then try to call it with named parameters. But that is not the case here. 

Named parameters generally work with magic static methods, see: https://3v4l.org/jOs8T

I think the problem here is instead that the parameters are accessed via a numeric key instead of their name, e.g.: https://github.com/pimcore/pimcore/blob/cc1a1fff78686423f616c7fe7c5d634a2ac71600/models/DataObject/Concrete.php#L602-L605

Which means the code is just not ready for named parameters yet.

**Comment by youwe-petervanderwal** [comment]
Indeed, I didn't get to the point to create a fix (yet). Might be that other static methods are affected too and of course it needs some proper testing (also to ensure that the old method call notation isn't affected by changing the code)

A `var_dump` on `$arguments` displayed a `array('limit' => 1)` so I expect that a 
```
$limit = $arguments['limit'] ?? $arguments[1] ?? null; 
```
would solve the problem

**Comment by jdreesen** [comment]
I think this should be reopened and fixed, since named parameters are a thing in PHP 8 and as far as I know Pimcore nowhere says that you should **not** use them.
